cmd/metal-api/internal/service/machine-service.go

```diff
@@ -628,6 +628,16 @@ func (r *machineResource) setMachineState(request *restful.Request, response *re
 		return
 	}
 
+	user, err := r.userGetter.User(request.Request)
+	if err != nil {
+		r.sendError(request, response, defaultError(err))
+		return
+	}
+	var userId string
+	if user != nil {
+		userId = user.EMail
+	}
```

Comment:
You can assume the user is not nil when there was no `err` returned.

Reply:
Thanks! Removed the check and the corresponding unit test